pubspec.yaml

```diff
@@ -12,6 +12,7 @@ dependencies:
   meta: ^1.1.7
 
 dev_dependencies:
+  charcode: ^1.3.0
```

Comment:
Don't depend on charcode! it's not a core Dart package!

You can use `package:charcode` to generate the constants you need instead.

Reply:
Why not? This isn't even a normal dependency, it's only used for tests.

Reply:
Dart-core packages should not depend on non-dart-core packages (unless absolutely necessary and with good arguments).
If we do another breaking change like null safety, we don't want our tests to be stuck on waiting for migration of a package that doesn't have similar support promises to the the core package itself. Even if it's only a dev-dependency, we'd still potentially be held back in migrating the tests, which are just as important as the rest of the package.